Update style_dir()

jonmcalder opened the pull request:
Extension to `style_dir()` to address the first part of #291

lorenzwalthert commented on R/ui.R:
```diff
@@ -100,10 +104,11 @@ style_dir <- function(path = ".",
                       ...,
                       style = tidyverse_style,
                       transformers = style(...),
+                      filetype = "R",
```
can we change that to `c("r")` also?

jonmcalder replied:
Sorry did the other merge from my phone and I thought you'd changed it already. Happy to change this too.

lorenzwalthert replied:
@krlmlr do you prefer to stick with `"R"` everywhere instead of `"r"`? Its kind of a philosophical debate I guess...

krlmlr replied:
Can we use `"R"` everywhere? It doesn't seem to affect the code, but the default file extension is `.R` and not `.r`, and we should be consistent here.

jonmcalder replied:
I agree that the default here and elsewhere should be uppercase. I think @lorenzwalthert wanted to emphasize that this argument can be/is a vector so wanted to use c("R") instead. I think the case inconsistency was incidental/accidental.

krlmlr replied:
I thought `c("R")` was an artifact... I feel it's enough if we just mention that the argument also accepts vectors in the documentation.

lorenzwalthert replied:
ok, fine.

lorenzwalthert commented on R/ui.R:
```diff
@@ -36,6 +39,7 @@ style_pkg <- function(pkg = ".",
                       ...,
                       style = tidyverse_style,
                       transformers = style(...),
+                      filetype = c("r"),
```
I don't see how this argument does affect the function call, i.e. it is not really used further downstream if I see that correctly.

lorenzwalthert replied:
So was your intend to wait with merging anyways until #293 is resolved?

jonmcalder replied:
Yeah this was just because the other functions (style_dir & prettify_any) were already inheriting params from here and I figured we'd be adding this to style_pkg later anyway.

lorenzwalthert replied:
Ok, lets wait with merging then.

jonmcalder commented:
Ok, this should be ready for further review now.

krlmlr commented on R/ui.R:
```diff
@@ -18,6 +18,9 @@ NULL
 #'   conveniently constructed via the `style` argument and `...`. See
 #'   'Examples'.
+#' @param filetype Vector of file extensions indicating which filetypes should
+#'   be styled. Case is ignored, and the `.` is optional, e.g. c(".R", ".Rmd")
```
Does the code need backticks?

krlmlr commented on R/set-assert-args.R:
```diff
@@ -0,0 +1,73 @@
+    stop(
+      "Token(s) ", paste0(invalid_tokens, collapse = ", "), " are invalid. ",
+      "You can lookup all valid tokens and their text ",
+      "with styler:::looup_tokens(). Make sure you supply the values of ",
```
looup -> lookup

krlmlr commented on R/ui.R:
```diff
@@ -37,24 +40,47 @@ style_pkg <- function(pkg = ".",
+      path = "vignettes", pattern = "\\.rmd$", ignore.case = TRUE,
+      recursive = TRUE, full.names = TRUE
+    )
+    readme <- dir(pattern = "readme\\.rmd", ignore.case = TRUE)
```
Needs `^` and `$` ?

krlmlr commented on R/ui.R:
```diff
@@ -37,24 +40,47 @@ style_pkg <- function(pkg = ".",
+  r_files <- test_files <- data_raw_files <- vignette_files <- readme <- NULL
+
+  if ("\\.r" %in% filetype) {
+    r_files <- dir(
```
The `paths` argument to `dir()` seems to be vectorized, not sure about older versions of R.

jonmcalder replied:
Thanks - I had overlooked this. Based on [this source code from 2000](https://github.com/wch/r-source/blob/92142731348d937ec347a56a8b68d34e9dc1d172/src/library/base/man/list.files.Rd), I think it's safe to assume that the paths argument has been vectorized since pre `R 2.0` days, so I'll go ahead and combine these.

krlmlr commented on R/ui.R:
```diff
@@ -37,24 +40,47 @@ style_pkg <- function(pkg = ".",
-  r_files <- dir(
-    path = "R", pattern = "[.][rR]$", recursive = TRUE, full.names = TRUE
-  )
+prettify_local <- function(transformers, filetype, exclude_files) {
```
I'm not sure the function name still matches the intent.

krlmlr commented on tests/testthat/public-api/xyz-r-and-rmd-dir/subdir/random-rmd-script-in-subdir.Rmd:
````diff
@@ -0,0 +1,30 @@
+---
+
+```{r setup, include=FALSE}
+1+1
````
Do the tests actually test that styling is applied?

lorenzwalthert replied:
I think we just check whether the file was named in the message about changed files. Should we test for the actual change?

jonmcalder replied:
They don't check whether styling is applied, just which files are being named in the changed files message.  @lorenzwalthert suggested these tests so I kinda assumed we had our bases covered with actual styling tests elsewhere.

jonmcalder commented:
Thanks very much for the review @krlmlr! Unfortunately I won't be able to get back to this until this evening (i.e. outside office hours). I hope I'm not holding up your CRAN release?

krlmlr commented:
No, that's totally fine.

krlmlr commented:
Thanks for working on this!

jonmcalder commented:
Hi @krlmlr! Sorry I should have requested a follow up review from you on Wednesday eve after pushing some changes.

I think I have addressed all your comments. The drop in testing coverage is very marginal - but in any case highlights the question of whether or not you feel there is sufficient testing? (please refer to the comments above)

krlmlr commented:
Thanks again!

jonmcalder commented:
Happy to help out - another good learning opportunity for me. Thanks @krlmlr!
